### PR TITLE
fix(context-pad): clear hover timeout on close

### PR DESCRIPTION
**Pull request by philippfromme**
brave_FzqvDuhVLx

---

Related to bpmn-io/bpmn-js#2004

### CODE REVIEW

File: lib/features/context-pad/ContextPad.js [hunk 1]
```diff
@@ -442,6 +446,12 @@ ContextPad.prototype.close = function() {
     return;
   }
 
+  if (this._timeout) {
+    clearTimeout(this._timeout);
+
+    this._timeout = null;
```
Comment (marstamm) [inline lib/features/context-pad/ContextPad.js:452]: What is the reason we set the timeout to null? As I can test, `clearTimeout` works fine with an already canceled timeout, so we could reduce complexity in our logic and just cancel all the time

Comment (marstamm) [reply lib/features/context-pad/ContextPad.js:452]: I'm also fine with merging it as-is if it is intended

File: lib/features/context-pad/ContextPad.js [hunk 2]
```diff
@@ -235,7 +235,11 @@ ContextPad.prototype.trigger = function(action, event, autoActivate) {
       this._mouseout = this.triggerEntry(entry, 'hover', originalEvent, autoActivate);
     }, HOVER_DELAY);
   } else if (action === 'mouseout') {
-    clearTimeout(this._timeout);
+    if (this._timeout) {
+      clearTimeout(this._timeout);
+
+      this._timeout = null;
+    }
```
Comment (nikku) [inline lib/features/context-pad/ContextPad.js:242]:
```suggestion
    clearTimeout(this._timeout);
```

File: lib/features/context-pad/ContextPad.js [hunk 3]
```diff
@@ -442,6 +446,12 @@ ContextPad.prototype.close = function() {
     return;
   }
 
+  if (this._timeout) {
+    clearTimeout(this._timeout);
+
+    this._timeout = null;
+  }
```
Comment (nikku) [inline lib/features/context-pad/ContextPad.js:453]:
```suggestion
  clearTimeout(this._timeout);
```

**Pull request merged**